proj/src/main/webapp/chat.js

```diff
@@ -924,6 +924,35 @@ function addUserInfoToDom(userObj) {
     }
 }
 
+function initDeleteAccountButton(){
+    const txtEmail = document.getElementById("email");
+    const txtPassword = document.getElementById("pass");
+    const btnDelete = document.getElementById("btnDeleteAccount");
+    
+    
+
+    btnDelete.addEventListener("click", async function () {
+        const emailVal = txtEmail.value;
+        const passVal = txtPassword.value;
+
+        console.log(emailVal);
+
+        // Re-authenticate credentials
+        const auth = firebase.auth();
+        const promise = auth.signInWithEmailAndPassword(emailVal, passVal).then(function(){
+            // Deletes user from database but not their messages 
+            // A new account can be created with the previously associated email 
+            var user = firebase.auth().currentUser;
+            user.delete().then(function() {
+                window.location.replace(welcome.html);
+            }).catch(function(error) {
+                error => alert(error.message);
+            });
+        });
+        promise.catch(e => alert(e.message))
```

Comment:
This deletes the user from authentication, can you additionally remove the user from the database and clear the user's uid from the chat?

Reply:
i deleted the user uid but though to kept their old messages for the sake of chat history